src/Parser/Priest/Discipline/Modules/Core/ShadowfiendAbilityTracker.js

```diff
@@ -0,0 +1,18 @@
+import SPELLS from 'common/SPELLS';
+
+import CoreAbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+
+class ShadowfiendAbilityTracker extends CoreAbilityTracker {
+
+  on_byPlayer_cast(event) {
+    if (super.on_byPlayer_cast) {
+      super.on_byPlayer_cast(event);
+    }
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.SHADOWFIEND_WITH_GLYPH_OF_THE_SHA.id || spellId === SPELLS.LIGHTSPAWN.id) {
+      event.ability.guid = SPELLS.SHADOWFIEND.id;
```

Comment:
How about making a Normalizer for this instead? I wouldn't expect this kind of behavior in an Analyzer and I think it will be hard to find for others, and it won't be shown in the events tab.